src/overlays/actors/ovl_En_Hgo/z_en_hgo.c

```diff
@@ -196,140 +217,161 @@ void EnHgo_DefaultDialogueHandler(EnHgo* this, PlayState* play) {
+                    this->animIndex = HGO_ANIM_CONSOLE;
                     Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, 3);
                     break;
+
                 case 5:
-                    this->unk_218 = 4;
+                    this->animIndex = HGO_ANIM_CONSOLE_HEAD_UP;
                     Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, 4);
                     break;
+
                 case 6:
-                    this->unk_218 = 5;
+                    this->animIndex = HGO_ANIM_REACH_DOWN_TO_LIFT;
                     Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, 5);
                     break;
             }
         } else if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
-            switch (this->unk_218) {
-                case 1:
-                    if ((Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) && (this->unk_312 == 0)) {
-                        this->unk_312 = 1;
+            switch (this->animIndex) {
+                case HGO_ANIM_ASTONISHED:
+                    if ((Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) &&
+                        (this->isInCutscene == false)) {
+                        this->isInCutscene = true;
                         if ((gSaveContext.sceneLayer == 0) &&
                             ((play->csCtx.currentCsIndex == 2) || (play->csCtx.currentCsIndex == 4))) {
                             Actor_PlaySfxAtPos(&this->actor, NA_SE_VO_GBVO02);
                         }
                     }
                     break;
-                case 2:
-                    this->unk_218 = 3;
+
+                case HGO_ANIM_KNEEL_DOWN_AND_HUG:
+                    this->animIndex = HGO_ANIM_CONSOLE;
                     Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, 3);
                     break;
-                case 5:
-                    this->unk_218 = 6;
+
+                case HGO_ANIM_REACH_DOWN_TO_LIFT:
+                    this->animIndex = HGO_ANIM_TOSS;
                     Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, 6);
             }
         }
 
         Cutscene_ActorTranslateAndYaw(&this->actor, play, actionIndex);
         return true;
     }
+
     if ((play->csCtx.state == 0) && (((gSaveContext.save.weekEventReg[75]) & 0x20)) &&
-        (this->actionFunc == func_80BD0410)) {
+        (this->actionFunc == EnHgo_DoNothing)) {
         this->actor.shape.rot.y = this->actor.world.rot.y;
         Actor_Spawn(&play->actorCtx, play, ACTOR_ELF_MSG2, this->actor.focus.pos.x, this->actor.focus.pos.y,
                     this->actor.focus.pos.z, 7, 0, 0, 0x7F5A);
-        func_80BD0420(this);
+        EnHgo_SetupInitCollision(this);
     }
-    this->unk_316 = 0x63;
+
+    this->csAction = 0x63;
     return false;
 }
 
-void func_80BD0B8C(EnHgo* this, PlayState* play) {
+void EnHgo_FacePlayer(EnHgo* this, PlayState* play) {
```

Comment:
Feels like this is as much about the blink routine as looking at Player, name ought to reflect this.

Reply:
How about `EnHgo_UpdateFace`?

Reply:
Hmm, that misses the body part. Maybe just `UpdatePoseAndEyes`?